### PR TITLE
feat: My Collection search bar

### PR DESCRIPTION
**Pull request by olerichter00**
The type of this PR is: **Feature**

This PR resolves [CX-1989]

### Description

This PR adds a search bar to the "My Collection" screen. In addition to that, it extracts the functions to locally filter artworks and adds logic to locally filter by keyword. 

**Note:**  Please go through the commits to review the different changes.

https://user-images.githubusercontent.com/4691889/151990619-1ea2c0b2-9300-4326-b763-76a806e76aa9.mp4

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [ ] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests/stories for my changes, or my changes don't require testing/stories, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/main/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/main/docs/adding_state_migrations.md))
- [x] I have added a changelog entry below or my changes do not require one.

### To the reviewers 👀

- [ ] I would like at least one of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- Add My Collection search bar (behind feature flag) - ole

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

[CX-1989]: https://artsyproduct.atlassian.net/browse/CX-1989?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  ❌ **An error occurred while validating your changelog, please make sure you provided a valid changelog.**
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 3540dcd165f0a4fbae99f415bd6da0765f609eee
</p>

File: src/lib/Scenes/MyCollection/utils/localArtworkSortAndFilter.ts [hunk 1]
```diff
@@ -0,0 +1,315 @@
+import {
```
Comment (olerichter00) [inline src/lib/Scenes/MyCollection/utils/localArtworkSortAndFilter.ts:1]: I extracted all the logic for local artwork filtering into this file. Apart from adding a function to filter by keyword I haven't changed anything.

**Comment by rajsam003** [comment]
Great work @olerichter00 ,

**Thought**: I think it would be a good user experience if we show a text something like - `No artwork found` for 0 search results instead of a blank screen.

File: src/lib/Components/StickyTabPage/StickyTabPage.tsx [hunk 2]
```diff
@@ -76,8 +79,16 @@ export const StickyTabPage: React.FC<StickyTabPageProps> = ({
...
   const { jsx: stickyHeader, nativeHeight: stickyHeaderHeight } =
     useAutoCollapsingMeasuredView(stickyHeaderContent)
+
   const tracking = useTracking()
+
   const headerOffsetY = useAnimatedValue(0)
+
+  // We need to set the header offset to 0 after hiding or showing the static header.
+  useEffect(() => {
+    requestAnimationFrame(() => headerOffsetY.setValue(0))
```
Comment (lordkiz) [inline src/lib/Components/StickyTabPage/StickyTabPage.tsx:89]: Won't this show the back button when you have scrolled down on a page with lots of content and you expect the back button to remain hidden?

File: src/lib/Components/StickyTabPage/SitckyTabPageContext.tsx [hunk 3]
```diff
@@ -9,6 +9,8 @@ export const StickyTabPageContext = React.createContext<{
   tabLabels: string[]
   tabSuperscripts: Array<string | undefined>
   activeTabIndex: GlobalState<number>
+  showStaticHeader: () => void
+  hideStaticHeader: () => void
```
[the remark this answers is not in the document]
Comment (lordkiz) [reply src/lib/Components/StickyTabPage/SitckyTabPageContext.tsx:13]: 👍🏾 looks elegant to me

File: src/lib/Components/StickyTabPage/StickyTabPage.tsx (hunk 2 above)
Comment (olerichter00) [reply src/lib/Components/StickyTabPage/StickyTabPage.tsx:89]: Good catch! I've just checked and it works fine. This effect is just for the case where we need to hide the static header from outside of the component (as used on the My Collection screen to scroll to the input). Other screens should not be affected by this change.

**Comment by olerichter00** [comment]
> Great work @olerichter00 ,
> 
> **Thought**: I think it would be a good user experience if we show a text something like - `No artwork found` for 0 search results instead of a blank screen.

Good catch! I'll create a ticket for it because we need to clarify the copy.

**Review by lordkiz — APPROVED** [review]
LGTM 👍

File: src/lib/Scenes/MyCollection/Components/MyCollectionSearchBar.tsx [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+  onFocus,
+}) => {
+  const [value, setValue] = useState("")
+  const [isFocused, setIsFocused] = useState(false)
+
+  const { hideStaticHeader, showStaticHeader } = useStickyTabPageContext()
+
+  const inputRef = useRef<TextInput>(null)
+
+  const debouncedSetKeywordFilter = useMemo(() => debounce((text) => onChangeText?.(text), 200), [])
```
Comment (MounirDhahri) [inline src/lib/Scenes/MyCollection/Components/MyCollectionSearchBar.tsx:29]: **praise:** 👏

File: src/lib/Scenes/MyCollection/utils/localArtworkSortAndFilter.ts [hunk 5]
```diff
@@ -0,0 +1,315 @@
...
+  FilterParamName,
+} from "lib/Components/ArtworkFilter/ArtworkFilterHelpers"
+import { ArtworksFiltersStore } from "lib/Components/ArtworkFilter/ArtworkFilterStore"
+import { FilterDisplayConfig } from "lib/Components/ArtworkFilter/types"
+import { filter, orderBy, trim, uniqBy } from "lodash"
+import { DateTime } from "luxon"
+import { useEffect } from "react"
+import { MyCollectionArtworkEdge } from "../MyCollection"
+
+export const useLocalArtworkFilter = (artworks: any[]) => {
```
Comment (MounirDhahri) [inline src/lib/Scenes/MyCollection/utils/localArtworkSortAndFilter.ts:14]: **praise:** good thinking of extracting this to a separate file

**Pull request merged**